source/source.go

```diff
@@ -2,68 +2,37 @@ package source
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials"
 	"github.com/aws/aws-sdk-go-v2/service/sqs"
 	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
 	sdk "github.com/conduitio/conduit-connector-sdk"
-	cf "github.com/meroxa/conduit-connector-amazon-sqs/config"
 )
 
 type Source struct {
 	sdk.UnimplementedSource
-	config      cf.Config
-	svc         *sqs.Client
-	queueURL    *string
-	sqsMessages *sqs.ReceiveMessageOutput
+	config   Config
+	svc      *sqs.Client
+	queueURL string
 }
 
-const (
-	timeout = 12 * 60 * 60
-)
-
 func NewSource() sdk.Source {
-	return sdk.SourceWithMiddleware(&Source{})
+	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware()...)
 }
 
 func (s *Source) Parameters() map[string]sdk.Parameter {
-	return map[string]sdk.Parameter{
-		cf.ConfigKeyAWSAccessKeyID: {
-			Default: "",
-			Validations: []sdk.Validation{
-				sdk.ValidationRequired{},
-			},
-			Description: "AWS Access Key ID.",
-		},
-		cf.ConfigKeyAWSSecretAccessKey: {
-			Default: "",
-			Validations: []sdk.Validation{
-				sdk.ValidationRequired{},
-			},
-			Description: "AWS Secret Access Key.",
-		},
-		cf.ConfigKeyAWSToken: {
-			Default:     "",
-			Description: "AWS Access Token (optional).",
-		},
-		cf.ConfigKeyAWSQueue: {
-			Default: "",
-			Validations: []sdk.Validation{
-				sdk.ValidationRequired{},
-			},
-			Description: "AWS SQS Queue Name.",
-		},
-	}
+	return Config{}.Parameters()
 }
 
 func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
 	sdk.Logger(ctx).Debug().Msg("Configuring Source Connector.")
-	parsedCfg, err := cf.ParseConfig(cfg)
+
+	err := sdk.Util.ParseConfig(cfg, &s.config)
 	if err != nil {
-		return err
+		return fmt.Errorf(" failed to parse source config : %w", err)
```

Comment:
Nitpick: no need for a space before the message and before the colon.
```suggestion
		return fmt.Errorf("failed to parse source config: %w", err)
```